pkg/cloud_provider/file/file.go

```diff
@@ -504,55 +533,59 @@ func (manager *gcfsServiceManager) ResizeInstance(ctx context.Context, obj *Serv
-		return nil, status.Error(codes.InvalidArgument, err.Error())
-	}
+func (manager *gcfsServiceManager) CreateBackup(ctx context.Context, backupInfo *BackupInfo) (*filev1beta1.Backup, error) {
 
-	backupSource := fmt.Sprintf("projects/%s/locations/%s/instances/%s", obj.Project, obj.Location, obj.Name)
 	backupobj := &filev1beta1.Backup{
-		SourceInstance:  backupSource,
-		SourceFileShare: obj.Volume.Name,
+		SourceInstance:  backupInfo.BackupSource(),
+		SourceFileShare: backupInfo.SourceShare,
+		Labels:          backupInfo.Labels,
 	}
-	klog.V(4).Infof("Creating backup object %+v for the URI %v", *backupobj, backupUri)
-	opbackup, err := manager.backupService.Create(locationURI(obj.Project, region), backupobj).BackupId(backupName).Context(ctx).Do()
+	klog.V(4).Infof("Creating backup object %+v for the URI %v", *backupobj, backupInfo.BackupURI)
+	opbackup, err := manager.backupService.Create(locationURI(backupInfo.Project, backupInfo.Location), backupobj).BackupId(backupInfo.Name).Context(ctx).Do()
 
 	if err != nil {
 		klog.Errorf("Create Backup operation failed: %w", err)
 		return nil, err
 	}
 
-	klog.V(4).Infof("For backup uri %s, waiting for backup op %v to complete", backupUri, opbackup.Name)
+	klog.V(4).Infof("For backup uri %s, waiting for backup op %v to complete", backupInfo.BackupURI, opbackup.Name)
 	err = manager.waitForOp(ctx, opbackup)
 	if err != nil {
-		return nil, fmt.Errorf("WaitFor CreateBackup op %s for source instance %v, backup uri: %v, operation failed: %w", opbackup.Name, backupSource, backupUri, err)
+		return nil, fmt.Errorf("WaitFor CreateBackup op %s for source instance %v, backup uri: %v, operation failed: %w", opbackup.Name, backupInfo.BackupSource(), backupInfo.BackupURI, err)
 	}
 
-	backupObj, err := manager.backupService.Get(backupUri).Context(ctx).Do()
+	backupObj, err := manager.backupService.Get(backupInfo.BackupURI).Context(ctx).Do()
 	if err != nil {
 		return nil, err
 	}
 	if backupObj.State != "READY" {
-		return nil, fmt.Errorf("backup %v for source %v is not ready, current state: %v", backupUri, backupSource, backupObj.State)
+		return nil, fmt.Errorf("backup %v for source %v is not ready, current state: %v", backupInfo.BackupURI, backupInfo.BackupSource(), backupObj.State)
 	}
-	klog.Infof("Successfully created backup %+v for source instance %v", backupObj, backupSource)
+	klog.Infof("Successfully created backup %+v for source instance %v", backupObj, backupInfo.BackupSource())
 	return backupObj, nil
 }
 
+func (manager *gcfsServiceManager) CreateMultishareBackup(ctx context.Context, backupInfo *BackupInfo) (*filev1beta1.Backup, error) {
+
+	return manager.CreateBackup(ctx, backupInfo)
+}
+
+func (manager *gcfsServiceManager) CreateSingleshareBackup(ctx context.Context, backupInfo *BackupInfo) (*filev1beta1.Backup, error) {
+	return manager.CreateBackup(ctx, backupInfo)
+}
```

Comment:
why do we need separate functions when the content are the same?

Reply:
Oh, I forgot to delete the bodies here. The content was different before refactor, and they aren't exported anymore. Good catch